packages/common-cosmos/src/project/models.ts

```diff
@@ -52,6 +60,9 @@ export class CosmosMessageFilter implements SubqlCosmosMessageFilter {
   @IsOptional()
   @IsString()
   contractCall?: string;
+  @IsOptional()
+  @Type(() => CosmosTxFilter)
+  txFilter?: CosmosTxFilter;
```

Comment:
Rather than nesting this I think that we can have `CosmosMessageFilter extends CosmosTxFilter`.
Example here https://github.com/subquery/subql/blob/main/packages/common-substrate/src/project/models.ts#L47